kolibri/plugins/coach/assets/src/composables/useQuizCreation.js

```diff
@@ -115,7 +126,24 @@ export default function useQuizCreation() {
     });
   }
 
-  function addQuestionsToSectionFromResources({ sectionIndex, resourcePool, questionCount }) {
+  function _insertItemsAt(baseArray, itemsToInsert, index) {
```

Comment:
Did you consider using [`splice`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice#syntax) instead of making a function? For whatever is passed in as `baseArray` you could do `baseArray.splice(index, 0, itemsToInsert)` - although it does mutate the array in place.

Not blocking as this looks good overall and I see you're doing checks on the passed index as well, it just got me thinking

Reply:
Oh I didnt :sweat_smile:. I just refactored the function to use the `splice` making a copy of the `baseArray` instead and it looks so much cleaner. Thanks @nucleogenesis!